test/groovy/com/sap/piper/UtilsTest.groovy

```diff
@@ -45,4 +61,39 @@ class UtilsTest {
 
         assert utils.getMandatoryParameter(parameters, 'test', null) == 'value'
     }
+
+    @Test
+    void testSWAReporting() {
+        utils.env = [BUILD_URL: 'something', JOB_URL: 'nothing']
+        utils.pushToSWA([step: 'anything'], [collectTelemetryData: true])
+        // asserts
+        assertThat(jscr.shell, hasItem(containsString('curl -G -v "https://webanalytics.cfapps.eu10.hana.ondemand.com/tracker/log"')))
+        assertThat(jscr.shell, hasItem(containsString('action_name=Piper Library OS')))
+        assertThat(jscr.shell, hasItem(containsString('custom3=anything')))
+        assertThat(jscr.shell, hasItem(containsString('custom5=`echo -n \'something\' | sha1sum | sed \'s/ -//\'`')))
+    }
+
+    @Test
+    void testDisabledSWAReporting() {
+        utils.env = [BUILD_URL: 'something', JOB_URL: 'nothing']
+        utils.pushToSWA([step: 'anything'], [collectTelemetryData: false])
+        // asserts
+        assertThat(jlr.log, containsString('[anything] Telemetry Report to SWA disabled!'))
```

Comment:
For all the tests checking the details for disabled SWAReporting: checking the corresponding log message is nice, but what about checking the shell calls for the callback url? The callback url must not be contained in any shell statement.

Reply:
done